coreneuron/network/partrans_setup.cpp

```diff
@@ -206,7 +206,7 @@ void nrn_partrans::gap_mpi_setup(int ngroup) {
 #if DEBUG
```

Comment:
Commit message in a3fbd5c summarise the fixes (actual fix is just few lines of code and didn't take long).
If we would have bit **structured information** about what/when/how fails then I would have jumped earlier here to help/fix the issues. For example:

* `-DCORENRN_DEBUG -DINTERLEAVE_DEBUG` was not giving any errors / working fine. 
* `INTERLEAVE_DEBUG` was failing in `mk_cell_indices()`. Ring test was working fine with cell-permute=1 but with 0 & 2 only. The [comment here](https://github.com/BlueBrain/CoreNeuron/blob/d2f4d3d87d390b08993f8f6f7fcbe606d0000b43/coreneuron/permute/cellorder.cpp#L359) indicated that it's for "cell per core style" meaning only permute = 1. 
* After fixing above, `-DINTERLEAVE_DEBUG` worked fine.
* The "real bug" existed with `-DCHKPNTDEBUG`. By looking at the assert failure for `assert(ntc.bcptype[i] == type)`, it was not difficult to find bug because `ntc.bcptype[i]` was only set/used at two places.